src/hazelweb/model/Program.re

```diff
@@ -370,17 +105,34 @@ let get_elaboration = (program: t): DHExp.t =>
   };
 
 exception EvalError(EvaluatorError.t);
-let evaluate = Memo.general(~cache_size_bound=1000, Evaluator.evaluate);
-let get_result = (program: t): Result.t =>
+exception PostprocessError(EvalPostprocessError.t);
+let (ec_init, env_init) = EvalEnv.empty;
+let evaluate =
+  Memo.general(
+    ~cache_size_bound=1000,
+    Evaluator.evaluate(ec_init, env_init),
+  );
+let get_result = (program: t): Result.t => {
   switch (program |> get_elaboration |> evaluate) {
-  | BoxedValue(d) =>
-    let (d_renumbered, hii) = renumber([], HoleInstanceInfo.empty, d);
-    (d_renumbered, hii, BoxedValue(d_renumbered));
-  | Indet(d) =>
-    let (d_renumbered, hii) = renumber([], HoleInstanceInfo.empty, d);
-    (d_renumbered, hii, Indet(d_renumbered));
+  | (es, BoxedValue(d)) =>
+    let (hci, d) =
+      switch (d |> EvalPostprocess.postprocess) {
+      | d => d
+      | exception (EvalPostprocessError.Exception(reason)) =>
```

Comment:
This doesn't actually catch any exceptions because the ones thrown in `EvalPostprocess` are `EvalPostprocess.Exception`, not `EvalPostprocessError.Exception`. The error/exception types are duplicated in the two modules---which should be kept?

Reply:
Fixed in https://github.com/hazelgrove/hazel/pull/586/commits/5018f013da425d9733a8f7c2f251c1f3c8f2a4e7 (see commit message).

My bad, didn't remove the `EvalPostprocessError` module when doing a refactor, I was meaning to move it to `EvalPostprocess.error`.